text/0000-pack-subcommands.md

```diff
@@ -0,0 +1,186 @@
+# Meta
+[meta]: #meta
+- Name: `pack` subcommands
+- Start Date: 2020-07-08
+- Author(s): @jromero
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: "N/A"
+
+# Summary
+[summary]: #summary
+
+Update the `pack` CLI to use more common subcommand patterns.
+
+# Motivation
+[motivation]: #motivation
+
+As the number of subcommands for `pack` grow, it's becoming more difficult to identify interworking subcommands at first glance. See the current list of subcommands below.
+
+```
+pack
+
+  build                 Generate app image from source code
+  rebase                Rebase app image with latest run image
+  inspect-image         Show information about a built image
+  set-run-image-mirrors Set mirrors to other repositories for a given run image
+  set-default-builder   Set default builder used by other commands
+  inspect-builder       Show information about a builder
+  suggest-builders      Display list of recommended builders
+  trust-builder         Trust builder
+  untrust-builder       Stop trusting builder
+  list-trusted-builders List Trusted Builders
+  create-builder        Create builder image
+  package-buildpack     Package buildpack in OCI format.
+  suggest-stacks        Display list of recommended stacks
+  version               Show current 'pack' version
+  report                Display useful information for reporting an issue
+  completion            Outputs completion script location
+  help                  Help about any command
+```
+
+# What it is
+[what-it-is]: #what-it-is
+
+
+### Proposed structure
+
+```
+pack
+
+  app
+    build
+    inspect
+    rebase
+  builder
+    create
+    inspect
+    suggest
+  buildpack
+    package
+    register
+  completion
+  config
+    default-builder
+      set
+      unset
+      show
+    run-image-mirror
+      add
+      remove
+      list
+    registry
+      add
+      remove
+      list
+    trusted-builder
+      add
+      remove
+      list
+  help
+  report
+  stack
+    suggest
+  version
+```
+
+### Top-Level Aliases
+
+There would also be some alias subcommands for the `app` set of subcommands for easier usage and compatibility.
+
+* `pack build` -> `pack app build`
+* `pack inspect` -> `pack app inspect`
```

Comment:
I like the `pack build` and `pack rebase` aliases but I think having a `pack inspect` that can't inspect builders would be confusing. I instead of making it smarter (which might fall apart if someday we can inspect things that aren't images) I think we should just remove this alias.

Reply:
Based on some discussions I think we want to move to where `pack inspect` can auto-detect the image resource type and display the appropriate information. For example, `pack inspect <builder>` will display builder info, `pack inspect <app-image>` will display app image information, etc.